### ISSUE TITLE
Events fail after Queue::fake()

### ISSUE THREAD

**Issue by apreiml** [laravel/framework#18223]
- Laravel Version: 5.3.30
- PHP Version: 7.0.9
- Database Driver & Version: MariaDB-10.1.16

### Description:

I'm testing, if jobs are queued using the fake queue. After I implemented throwing an event in one of my models the QueueFake fails to push.

```
Tests\Core\SomeTest::testQueueFake
ErrorException: get_class() expects parameter 1 to be object, string given

/home/ape/Symvaro/wasser_web/vendor/laravel/framework/src/Illuminate/Support/Testing/Fakes/QueueFake.php:135
/home/ape/Symvaro/wasser_web/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:423
/home/ape/Symvaro/wasser_web/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:354
/home/ape/Symvaro/wasser_web/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:221
/home/ape/Symvaro/wasser_web/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php:480
...
```

### Steps To Reproduce:

```
class CustomerChangeEvent
{
    use SerializesModels;

    public $someModel;

    ...
}

public function testQueueFake() {
    Queue::fake();

    event(CustomerChangeEvent::class);
}
```

**Comment by astierler** [comment]
@apreiml not sure if this would help or not, but can you try with adding `Event::fake()` as well?  Not sure if they both need to be done to play nicely together.

**Comment by astierler** [comment]
Actually, what's most likely happening is you're passing just the class name into `event()`

Try doing:  `event(new CustomChangeEvent());` since it's expecting an object on the first param, whereas ::class only returns the actual class name as a string, ie. `'CustomerChangeEvent'`

**Comment by apreiml** [comment]
@astierler Hi, thanks for the input. The class name should also be supported, according to the source comments. Also I don't want the Events to be faked, because it like to see if the event properly adds a job to the queue.

I checked this again and it seems to fixed with the current version of Laravel 5.4.*
